packages/core/admin/admin/src/content-manager/hooks/useLazyComponents/index.js

```diff
@@ -11,13 +11,13 @@ const componentStore = new Map();
  */
 const useLazyComponents = (componentUids = []) => {
   const [lazyComponentStore, setLazyComponentStore] = useState(Object.fromEntries(componentStore));
-  const [loading, setLoading] = useState(() => {
-    if (componentStore.size === 0 && componentUids.length > 0) {
-      return true;
-    }
+  /**
+   * Start loading only if there are any components passed in
+   * and there are some new to load
+   */
+  const [newUids] = useState(() => componentUids.filter((uid) => !componentStore.get(uid)));
+  const [loading, setLoading] = useState(() => !!newUids.length);
```

Comment:
I was thinking something like this:

```suggestion
  const newUids = componentUids.filter((uid) => !componentStore.get(uid));
  const [loading, setLoading] = useState(() => !!newUids.length);
```
I tested it locally and it seemed to be working fine

Reply:
I agree, if you don't need the `useState` setter then there's no reason to use state for `newUids`

Reply:
useMemo + a good comment explaining why it's there works for me, I just didn't see much of a difference without it when I tested so wasn't sure why it was necessary.

I am wondering though how the value of `newUids` would change after the component has mounted?

Reply:
if the reconciler doesn't think it should be disposed then it won't be unmounted e.g. we move the entities the reconciler may believe there are parts of the tree that are not invalid so to be safe, we should use a memo i believe :)

Reply:
@joshuaellis I'm not sure I fully understand this situation, but if we rely on memo to avoid a bug (and not just as a perf optimization), I think we have to be careful because React could flush the memo cache when it wants

Reply:
So i'm still here and just waiting for this discussion to complete. Is there any final decision on implementation already?

Reply:
If your code breaks because react flushes a memoisation, its a bad implementation. Right now (in the PR not marks comment) if the argument `componentUids` changes, it will not update, so either we we leave it as a derived value or it goes in memo but the same goes for the `loading` state object.